### DISCUSSION TITLE
건의사항

### DISCUSSION THREAD

**Discussion by kimseungki94** [byungmeo/GersangStation#40] [Announcements]
거상을 할 때, 해당 서비스를 사용하는 유저입니다 ^^
해당 서비스의 경우 기존에 불필요한 프로세스의 간소화가 목적이라고 봅니다.
그 목적이라는 관점에서 생각한다면, 상용화 서비스가 아니라면, 나중에
사용자의 휴대전화 번호를 받고, 휴대전화 번호를 바탕으로 OTP 사용자의 경우 알람으로 자동으로 OTP 어플에 들어갈 수 있도록 해보는거도 좋을듯 합니다..
물론 보안 이슈가 있어서 어렵다면 어쩔수 없지만요..
추가적으로 추후 괜찮으시다면, 이벤트 클릭 로그를 별도로 수집해서, 사용자에게 얼마나 이벤트를 참여햇는지나 혹은 참여해야된다는 알람을 보여주는거도 좋을듯 합니다...!

**Comment by byungmeo** [comment]
@kimseungki94 
안녕하세요 건의 주셔서 감사합니다.

안타깝게도 말씀주신 건의사항들은 언급하셨다시피 보안 이슈 때문에 구현이 어렵습니다 ㅜㅜ
경제적인 문제 때문에도 그렇구요.

배경을 조금 설명 드리자면
사실 패스워드를 자신의 컴퓨터에 저장 하는 동작부터 일반 사용자들의 입장에선 납득하기 어려운 상황인게 현실입니다.
깃허브에 코드를 오픈하긴 했지만, 언제든지 배포파일만 살짝 바꿔서 게시하면 사용자들의 패스워드를 훔치는 건 일도 아닐겁니다.
출시 초기부터 약 1여년 동안은 농담, 풍자가 대부분이겠지만 일부 커뮤니티에서는 해킹 툴이라는 언급도 가끔 들려왔구요..
이 부분에 대해 해명하는데 적지 않은 노력과 시간을 들인 것 같습니다.
지금은 그냥 많은 분들이 사용하니까 인지도 빨(?)로 이 부분에 대한 의심은 사그러진 것 같지만요,,

아시다시피 휴대전화 번호를 바탕으로 한 알림 서비스나 로그 기반의 알림 서비스는 결국 서버나 DB를 필요로 하는데,
애초에 제가 패스워드를 해킹하지 않는다는 근거 중 하나로 내세운 것이 거상 스테이션에는 서버 혹은 DB가 없다는 사실 입니다.
훔칠 쌀 저장할 창고도 안 지었으니 결백함을 인정해달라는 뜻이죠.
그래서 GitHub API를 통해서 공지 불러오고 업데이트 확인하는 방향으로 개발한겁니다.
학생이라 서버 유지할 돈도 없구요 ㅎㅎ

초기에는 출석체크 보상 수령 기한 임박 알림 같은 건의도 많이 고려해봤는데
이벤트 페이지나 방식이 매번 바뀌기도 하고,, 이것저것 고려할게 너무 많다보니
그냥 이벤트 페이지에 바로 접속할 수 있는 바로가기 버튼 만드는 정도로 만족을 하게 되었습니다.

주절주절 말이 많았는데
많은 관심 가져주시고 건의 주셔서 정말 감사하고
같은 학교 학우분 보니까 반갑네요 ㅎㅎ
혹시나 다른 질문이나 건의사항 있으시면 언제든지 말씀주세요 ^^

**Reply by kimseungki94** [reply]
@byungmeo 
**학생이라 서버 유지할 돈도 없구요 ㅎㅎ**
해당 부분에서 모든 이유가 납득이 된거같네요 ㅎㅎ
역시 보안적인 이슈가 있겠지만.. DB가 아니더라도 사용자 PC에 로그파일 혹은 텍스트 기반 파일을 생성하고, 
해당 파일 정보를 읽어서 처리하는게 아닌 이상 어려울거같네요..
그리고 설사 만들더라도, 결국 휴대전화번호는 개인정보이기 때문에 한계가 있겠군요..
좋은 서비스를 운영해주셔서 항상 감사드려요 ^^
